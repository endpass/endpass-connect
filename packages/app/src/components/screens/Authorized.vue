<template>
  <v-frame :loading="!inited">
    <logout-form 
      :loading="loading"
      :closable="isDialog"
      @submit="handleLogoutSubmit"
      @cancel="handleLogoutCancel"
    /> 
  </v-frame> 
</template>

<script>
import { mapActions, mapState, mapGetters } from 'vuex';
import VFrame from '../VFrame.vue';
import LogoutForm from '../LogoutForm.vue';

export default {
  name: 'Authorized',

  computed: {
    ...mapState({
      inited: state => state.core.inited,
      loading: state => state.core.loading,
    }),
    ...mapGetters(['isDialog']),
  },

  methods: {
    ...mapActions(['init', 'logout', 'cancelLogout', 'sendReadyMessage']),

    async handleLogoutSubmit() {
      this.logout();
    },

    handleLogoutCancel() {
      this.cancelLogout();
    },

    handleWindowClose() {
      this.cancelLogout();
    },
  },

  async created() {
    await this.init();

    if (this.isDialog) {
      window.addEventListener('beforeunload', this.handleWindowClose);

      await this.sendReadyMessage();
    }
  },

  components: {
    VFrame,
    LogoutForm,
  },
};
</script>
