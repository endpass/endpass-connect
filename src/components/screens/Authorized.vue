<template>
  <v-frame :loading="!inited">
    <logout-form 
      :loading="loading"
      :closable="isDialog"
      @submit="handleLogoutSubmit"
      @cancel="handleLogoutCancel"
    /> 
  </v-frame> 
</template>

<script>
import { mapActions, mapState, mapGetters } from 'vuex';
import VFrame from '../VFrame.vue';
import LogoutForm from '../LogoutForm.vue';

export default {
  name: 'Authorized',

  computed: {
    ...mapState({
      inited: state => state.core.inited,
      loading: state => state.core.loading,
    }),
    ...mapGetters(['isDialog']),
  },

  methods: {
    ...mapActions(['logout', 'cancelLogout']),

    async handleLogoutSubmit() {
      this.logout();
    },

    handleLogoutCancel() {
      this.cancelLogout();
    },

    handleWindowClose() {
      this.cancelLogout();
    },
  },

  async created() {
    if (this.isDialog) {
      window.addEventListener('beforeunload', this.handleWindowClose);
    }
  },

  components: {
    VFrame,
    LogoutForm,
  },
};
</script>
