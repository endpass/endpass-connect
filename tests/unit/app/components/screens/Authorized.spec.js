import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Authorized from '@@/app/src/components/screens/Authorized.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Authorized', () => {
  let store;
  let storeData;
  let wrapper;
  let accountsModule;
  let coreModule;

  beforeEach(() => {
    coreModule = {
      state: {
        inited: true,
        loading: false,
      },
      actions: {
        sendReadyMessage: jest.fn(),
      },
    };
    accountsModule = {
      state: {
        linkSent: false,
        accounts: null,
      },
      actions: {
        auth: jest.fn(),
        cancelAuth: jest.fn(),
        confirmAuth: jest.fn(),
        awaitAuthConfirm: jest.fn(),
        awaitAccountCreate: jest.fn(),
      },
    };
    storeData = {
      modules: {
        accounts: accountsModule,
        core: coreModule,
      },
    };
    store = new Vuex.Store(storeData);
    wrapper = shallowMount(Authorized, {
      localVue,
      store,
    });
  });

  describe('render', () => {
    it('should correctly render Authorized component', () => {
      expect(wrapper.name()).toBe('Authorized');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
