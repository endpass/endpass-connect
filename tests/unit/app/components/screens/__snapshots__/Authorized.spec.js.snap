// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorized render should correctly render Authorized component 1`] = `
<v-frame-stub>
  <logout-form-stub closable="true"></logout-form-stub>
</v-frame-stub>
`;
